src/core/tsi/ssl_transport_security_utils.cc

```diff
@@ -375,4 +380,79 @@ absl::StatusOr<std::string> AkidFromCrl(X509_CRL* crl) {
   return ret;
 }
 
+bool IsPemCertificateChainNonEmptyAndValid(absl::string_view cert_chain_pem) {
```

Comment:
This is a bit subtle because of what PEM is. These APIs are named if you're checking whether the input "is" a certificate chain, but non-PEM data and PEM blocks of the wrong type are typically skipped over when parsing. This means that this function returning true just tells you that it *contains* a certificate chain, not that it *is* a certificate chain.

Reply:
Thanks for calling this out!  I think we are fine skipping over any extra unrelated portions of the input string. The updated comment on the API also know no longer says that `cert_chain_pem` must consist of exactly the PEM cert chain.